From 46bdc8ceca42fd19cd2b97d9fa845860e537dee9 Mon Sep 17 00:00:00 2001
From: Chad Condon <chad@condon.tech>
Date: Fri, 27 Dec 2024 18:21:21 -0800
Subject: [PATCH 2/2] Squelch deprecation warning

Most recent CMake started complaining about the pending end of 3.8
support.

> ```
> CMake Deprecation Warning at CMakeLists.txt:4 (cmake_minimum_required):
>   Compatibility with CMake < 3.10 will be removed from a future version of
>   CMake.
>
>   Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
>   to tell CMake that the project requires at least <min> but has been updated
>   to work with policies introduced by <max> or earlier.
> ```

We can retain support by adding a max version. This will no
prevent use with newer versions, but indicates forward
compatibility.[^1]

[^1]: https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html

Signed-off-by: Moritz Haase <Moritz.Haase@bmw.de>
Upstream-Status: Backport [9dda473ded5913d47221e7ae50817dbcd45175ff]
---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d36c6491..0cc12cb1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,7 @@
 # 2.6.3 is needed for ctest support
 # 3.1 is needed for target_sources
-cmake_minimum_required(VERSION 3.1)
+# 3.8 is needed for try_compile improvements (CMP0067)
+cmake_minimum_required(VERSION 3.8...3.31)
 
 project(CppUTest)
 
